pkgs/development/compilers/dotnet/vmr.nix

```diff
@@ -276,6 +309,11 @@ in stdenv.mkDerivation rec {
     runHook postConfigure
   '';
 
+  postConfigure = lib.optionalString (lib.versionAtLeast version "9") ''
+    # see patch-npm-packages.proj
+    typeset -f isScript patchShebangs > src/aspnetcore/patch-shebangs.sh
```

Comment:
This feels wrong, but I couldn't think of a better way.

Reply:
Do these come from a package? we could instead copy `patch-npm-packages.proj` and then substitute with the path to the script

Reply:
I think referencing by path in nixpkgs would be better because those are less likely to change than their internal implementations, which would then be a silent breakage as we're depending on their implementation to pick what functions we put in `patch-shebangs.sh`.